src/context.rs

```diff
@@ -35,12 +35,11 @@ use crate::x::*;
 ///
 /// This callback must return 0 unless stated otherwise in the event
 /// description at [Event].
-pub type ContextCallback = dyn Fn(&Context, Event, uintptr_t, uintptr_t) -> uintptr_t;
+pub type ContextCallback = dyn Fn(&Context, Event, uintptr_t, uintptr_t) -> uintptr_t + Send + Sync;
 
 pub struct Context {
-    pub userdata: *mut libc::c_void,
-    pub dbfile: Arc<RwLock<Option<PathBuf>>>,
-    pub blobdir: Arc<RwLock<*mut libc::c_char>>,
+    pub(crate) dbfile: PathBuf,
+    pub(crate) blobdir: PathBuf,
```

Comment:
Why have them `pub(crate)` if you keep the accessor functions anyway?  Seems redundant and better to use one or the other?

Reply:
true